src/components/compress/index.tsx

```diff
@@ -301,7 +302,7 @@ export default class Compress extends Component<Props, State> {
       // The image only needs updated if the encoder/preprocessor settings have changed, or the
       // source has changed.
       if (sourceDataChanged || encoderChanged || preprocessorChanged) {
-        this.updateImage(i, {
+        this.compressionJobs[i] = this.updateImage(i, {
```

Comment:
I'm not sure these reflect "abort" correctly. I'm not sure what the right thing to do there is. Can comlink handle DOMException?